server/src/main/java/keywhiz/service/resources/StatusResource.java

```diff
@@ -26,12 +36,50 @@
 public class StatusResource {
   private Environment environment;
   private static final Logger logger = LoggerFactory.getLogger(SecretDeliveryResource.class);
+  LoadingCache<HealthCheckRegistry, SortedMap<String, HealthCheck.Result>> cachedResult;
 
-  @Inject public StatusResource(Environment environment) {
+  @Inject public StatusResource(KeywhizConfig keywhizConfig, Environment environment) {
     this.environment = environment;
+
+    Duration cacheExpiry = keywhizConfig.getStatusCacheExpiry();
+
+    cachedResult = CacheBuilder.newBuilder()
```

Comment:
Prefer `Suppliers.memoizeWithExpiration` instead of a cache: https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Suppliers.html